### PR TITLE
calib3d: add a calibration flag to fix tangential distortion

### LINKED ISSUE #8845: Feature Request: CV_CALIB_FIX_P1 and CV_CALIB_FIX_P2 flag for calibrateCamera
##### System information (version)

- OpenCV => Latest
- Operating System / Platform =>  Ubuntu 16.04
- Compiler => GCC 5.4

##### Detailed description

When specifying various flags when using the function calibrateCamera, according to http://docs.opencv.org/3.2.0/d9/d0c/group__calib3d.html#ga3207604e4b1a1758aa66acb6ed5aa65d there are options to fix K1 to K6, S1 to S4, but for some reason not P1 and P2. The only option is set them to zero. Is it possible to add in flags so we can provide an initial P1 and P2 and it will stay fixed during the optimization like all the other distortion coefficients?

### PR DESCRIPTION
**Pull request by sovrasov**
resolves #8845

### CODE REVIEW

File: modules/calib3d/src/calibration.cpp [hunk 1]
```diff
@@ -1498,24 +1498,26 @@ static double cvCalibrateCamera2Internal( const CvMat* objectPoints,
...
+    if( !(flags & CALIB_THIN_PRISM_MODEL))
         flags |= CALIB_FIX_S1_S2_S3_S4;
-    if( !(flags & CV_CALIB_TILTED_MODEL))
+    if( !(flags & CALIB_TILTED_MODEL))
         flags |= CALIB_FIX_TAUX_TAUY;
 
     mask[ 4] = !(flags & CALIB_FIX_K1);
     mask[ 5] = !(flags & CALIB_FIX_K2);
+    mask[6]  =
+    mask[7]  = !(flags & CALIB_FIX_TANGENT_DIST);
```
Comment (alalek) [inline modules/calib3d/src/calibration.cpp]: Looks like this line overrides changes from line 1508, so function behavior is changed.
What is about to guard with "if" condition?
```
if (flags & CALIB_FIX_TANGENT_DIST)
{
    mask[6]  = mask[7]  = 1;
}
```

Comment (sovrasov) [inline modules/calib3d/src/calibration.cpp]: @alalek fixed.

**Comment by alalek** [comment]
:+1:

**Comment by jasjuang** [comment]
@sovrasov @alalek I didn't realize this PR until now. I tested the CALIB_FIX_TANGENT_DIST but it is not working. I initialized my distortion parameters to `[0, 0, 0, 0, 0]`, and regardless of whether I set my flags to 

``````
cv::CALIB_FIX_K1 + cv::CALIB_FIX_K2 + cv::CALIB_FIX_K3 + cv::CALIB_FIX_TANGENT_DIST
``````
or
``````
cv::CALIB_FIX_K1 + cv::CALIB_FIX_K2 + cv::CALIB_FIX_K3
``````

The outcome is always `[0, 0, 0.003178797775587784, 0.003211421837904917, 0]` after I call `cv::calibrateCamera`

Were you guys able to get this flag to work?

**Pull request merged**